### PR TITLE
update code for sending notification email for practitioners

### PR DESCRIPTION
**Pull request by achen2401**
https://www.pivotaltracker.com/story/show/185407795
see [PR](https://github.com/uwcirg/isacc-messaging-client-sof/pull/55)
Patient's followers are now referenced on the `CareTeam` resource, referenced by the patient's `CarePlan`
hence the code for sending out emails to notify practitioners needs to be updated as well

### CODE REVIEW

File: isacc_messaging/api/isacc_record_creator.py [hunk 1]
```diff
@@ -179,24 +179,29 @@ def get_patient(self, patient_id):
...
+                        if result is not None:
+                            gp = Practitioner(result)
+                            for t in gp.telecom:
+                                if t.system == 'email':
+                                    emails.append(t.value)
         if not emails:
             isacc_messaging.audit.audit_entry(
-                "no practitioner email to notify",
-                extra={"Patient": str(pt)},
+                "no practioner email to notify",
```
Comment (mcjustin) [inline isacc_messaging/api/isacc_record_creator.py]: minor typo

**Review by mcjustin — APPROVED** [review]
thanks - just a minor typo

File: isacc_messaging/api/isacc_record_creator.py [hunk 2]
```diff
@@ -179,24 +179,31 @@ def get_patient(self, patient_id):
         if result is not None:
             return Patient(result)
 
-    def get_general_practitioner_emails(self, pt: Patient) -> list:
+    def get_practitioners_emails(self, patient_id) -> list:
```
Comment (pbugni) [inline isacc_messaging/api/isacc_record_creator.py]: @hannahburkhardt did a nice job of adding type annotations, let's maintain them.

make that:
```def get_practitioners_emails(self, patient_id: int) -> list:```

File: isacc_messaging/api/isacc_record_creator.py [hunk 3]
```diff
@@ -179,24 +179,31 @@ def get_patient(self, patient_id):
...
-                    gp = Practitioner(result)
-                    for t in gp.telecom:
-                        if t.system == 'email':
-                            emails.append(t.value)
+        care_plan = self.get_careplan(patient_id)
+        if care_plan and care_plan.careTeam and len(care_plan.careTeam) > 0:
+            # get the referenced CareTeam resource from the care plan
+            # please see https://www.pivotaltracker.com/story/show/185407795
+            # carePlan.careTeam now includes those that follow the patient
+            resource_type, resource_id = care_plan.careTeam[0].reference.split('/')
```
Comment (pbugni) [inline isacc_messaging/api/isacc_record_creator.py:195]: if `len(care_plan.careTeam) > 1` we'll silently ignore them.  perhaps this is okay?

**Review by pbugni — APPROVED** [review]
minor change request - looks great, thank you for doing this.

File: isacc_messaging/api/isacc_record_creator.py (hunk 3 above)
Comment (achen2401) [reply isacc_messaging/api/isacc_record_creator.py:195]: thanks @pbugni  this will be logged now if happens.

**Review by mcjustin — APPROVED** [review]

**Review by ivan-c — APPROVED** [review]

**Pull request merged**